ganttproject/src/net/sourceforge/ganttproject/GPTreeTableBase.java

```diff
@@ -528,10 +544,139 @@ protected void initTreeTable() {
     doInit();
   }
 
+  private class SortTableHeaderRenderer implements TableCellRenderer {
+    private final Icon myAscIcon;
+    private final Icon myDescIcon;
+    private final TableCellRenderer myDefaultRenderer;
+    SortTableHeaderRenderer(JTable t) {
+      myDefaultRenderer = t.getTableHeader().getDefaultRenderer();
+      myAscIcon = UIManager.getIcon("Table.ascendingSortIcon");
+      myDescIcon = UIManager.getIcon("Table.descendingSortIcon");
+    }
+
+    @Override
+    public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int col) {
+      Component c = myDefaultRenderer.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, col);
+      ColumnImpl column = myTableHeaderFacade.findColumnByViewIndex(col);
+      if (column.getSort() == SortOrder.ASCENDING) {
+        if (c instanceof JLabel) {
```

Comment:
Please move `instanceof` check upper. Do it prior to dealing with columns. This will make code cleaner, just by removing multiple nested ifs.